### PR TITLE
UX: improves arrow support in chat emoji picker

### PR DESCRIPTION
**Pull request by jjaffeux**

### CODE REVIEW

File: plugins/chat/assets/javascripts/discourse/components/chat-emoji-picker.js [hunk 1]
```diff
@@ -224,51 +238,82 @@ export default class ChatEmojiPicker extends Component {
...
-        sectionEmojis[sectionEmojis.length - 1].focus();
+      if (prevEmoji) {
+        prevEmoji.focus();
       } else {
-        event.target.previousElementSibling?.focus();
+        const prevSection = currentSection.previousElementSibling;
+        if (prevSection) {
+          focusSectionsLastEmoji(prevSection);
+        } else {
+          focusFilter();
```
Comment (keegangeorge) [inline plugins/chat/assets/javascripts/discourse/components/chat-emoji-picker.js:282]: Getting an error here:
```javascript
Uncaught TypeError: focusFilter is not a function
    at ChatEmojiPicker.didNavigateSection (chat-emoji-picker.js:228:1)
```
I believe it should be:
```suggestion
          focusFilter;
```

File: plugins/chat/assets/javascripts/discourse/components/chat-emoji-picker.js [hunk 2]
```diff
@@ -224,51 +238,82 @@ export default class ChatEmojiPicker extends Component {
...
 
-      sectionEmojis
+      const prevEmoji = allEmojis()
         .reverse()
         .filter((c) => c.offsetTop < event.target.offsetTop)
-        .find((c) => c.offsetLeft === event.target.offsetLeft)
-        ?.focus();
+        .findBy("offsetLeft", event.target.offsetLeft);
+
+      prevEmoji?.focus() || focusFilter();
```
Comment (keegangeorge) [inline plugins/chat/assets/javascripts/discourse/components/chat-emoji-picker.js:316]: Same issue here, I believe it should be:

```suggestion
      prevEmoji?.focus() || focusFilter;
```

**Review by keegangeorge — COMMENTED** [review]
This is looking great! Just tiny issue I've made a suggestion to fix above.

Also, are the other keyboard actions in separate PRs? The <kbd>Enter</kbd> to select an emoji and <kbd>Tab</kbd> to jump sections don't seem to work on this PR.

File: plugins/chat/assets/javascripts/discourse/components/chat-emoji-picker.js (hunk 2 above)
Comment (jjaffeux) [reply plugins/chat/assets/javascripts/discourse/components/chat-emoji-picker.js:316]: weird... not it's the opposite, focusFilter should be made a function, will change :+1:

**Review by keegangeorge — APPROVED** [review]

**Pull request merged**